### DISCUSSION TITLE
assembly aka extern aliases

### DISCUSSION THREAD

**Discussion by aBaechtold** [icsharpcode/ILSpy#2441] [Development] [answered]
Hi,

I was recently reviewing an assembly that relies on assembly aliases  (see also [extern alias (C# Reference)](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/extern-alias)) to access types in assemblies having the same fully qualified name but of different version. I noticed that ILSpy displays the types of the different versions to look exactly the same i.e. couldn't tell that they were different while they were. Clicking each of them with the mouse pointer navigated to the correct version, so that is good.

I wondered whether this could be an improvement and before submitting an issue wanted to ask if others think similarly. I did not find an issue or discussion that seems to cover this topic already, hence the new discussion. It's nothing critical as using assembly aliases is not a common thing I guess. It's also not showing something incorrectly, it's just ambiguity in the UI.

### Details:
ILSpy version 7.1.0.6543
.NET version 4.8.4300.0
 
**ILSpy disassembly:**
```
private void _GenerateListOfConfigModels()
	{
		List<VConfigInfo> list = new List<VConfigInfo>
		{
			new VConfigInfo
			{
				VersionString = new VConfig().ExpectedConfigVersion,
				ConfigType = typeof(VConfig)   //<=Here...
			},
			new VConfigInfo
			{
				VersionString = new VConfig().ExpectedConfigVersion,
				ConfigType = typeof(VConfig)  //<=and here
			}
		};
		...
	}
```

**Actual code:**
```
...
extern alias v1;
extern alias v2;
...
using VConfig_v1               = v1::VConfig;
using VConfig_v2               = v2::VConfig;
...
private void _GenerateList() 
   {
      List<VConfigInfo> list = new List<VConfigInfo>
          {
               new VConfigInfo {VersionString = new VConfig_v1().ExpectedConfigVersion, ConfigType = typeof(VConfig_v1)},
               new VConfigInfo {VersionString = new VConfig_v2().ExpectedConfigVersion, ConfigType = typeof(VConfig_v2)}
                ...
          };
     ...
    }
```

**Comment by siegfriedpammer** [comment, marked as answer]
I have created #2442 to track this. Thank you for the suggestion!
